### PR TITLE
[LegacyCard] Consolidate se23 styles and logic

### PR DESCRIPTION
**Pull request by sophschneider**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/9944

### WHAT is this pull request doing?

* Consolidate se23 logic and styles for the `LegacyCard` Component

### How to 🎩
* Compare production and this PR's chromatic storybook to make sure styles are the same

[Production](https://storybook.polaris.shopify.com/?path=/story/all-components-legacycard--all&globals=polarisSummerEditions2023:true)
[This PR](https://5d559397bae39100201eedc1-uabmwvpwiv.chromatic.com/?path=/story/all-components-legacycard--all)

### CODE REVIEW

**Comment by sophschneider** [comment]
Baseline UI chromatic changes look unrelated to this PR and there aren't actually any visual differences between this PR's storybook and the prod one. Also accepted on `next`

**Review by laurkim — APPROVED** [review]
Nice 👏 

Noticed a small spacing issue between actions on the [With Destructive Action](https://5d559397bae39100201eedc1-uabmwvpwiv.chromatic.com/?path=/story/all-components-legacycard--with-destructive-action) story but other than that the other stories looks great at all breakpoints 👍

File: polaris-react/src/components/LegacyCard/LegacyCard.scss [hunk 1]
```diff
@@ -31,48 +29,32 @@
...
-        $boxShadow: none,
-        $borderRadius: var(--p-border-radius-0-experimental),
-        $border: none,
-        $content: none
-      );
-    }
+    @include shadow-bevel(
+      $boxShadow: none,
+      $borderRadius: var(--p-border-radius-0-experimental),
+      $border: none,
```
Comment (laurkim) [inline polaris-react/src/components/LegacyCard/LegacyCard.scss:55]: I think these two args are passed into the mixin to ignore the [shadow bevel opt-out](https://github.com/Shopify/polaris/pull/10167/files/2cf64e80253c5a314d58601fd1216fa6803c63b0#r1301993816) and can be removed.

File: polaris-react/src/components/LegacyCard/LegacyCard.scss [hunk 2]
```diff
@@ -31,48 +29,32 @@
...
+      $content: none
```
Comment (laurkim) [inline polaris-react/src/components/LegacyCard/LegacyCard.scss:56]: I think the `$border` and `$content` args are passed into the mixin to ignore the [shadow bevel opt-out](https://github.com/Shopify/polaris/pull/10167/files/2cf64e80253c5a314d58601fd1216fa6803c63b0#r1301993816) and can be removed.

Comment (sophschneider) [reply polaris-react/src/components/LegacyCard/LegacyCard.scss:56]: I think I have to keep it! its to remove the shadow bevel in print

|Without|With|
|-|-|
|Screenshot 2023-08-24 at 2 01 27 PM|Screenshot 2023-08-24 at 2 01 44 PM|

Comment (laurkim) [reply polaris-react/src/components/LegacyCard/LegacyCard.scss:56]: Ooh ok good catch, I didn't test it so was 100% an assumption on my part just based off previous PR 😅

**Comment by sophschneider** [comment]
> Noticed a small spacing issue between actions on the [With Destructive Action](https://5d559397bae39100201eedc1-uabmwvpwiv.chromatic.com/?path=/story/all-components-legacycard--with-destructive-action) story but other than that the other stories looks great at all breakpoints 👍

@laurkim nice catch!!! It actually has the same spacing in the most recent `next` [storybook](https://5d559397bae39100201eedc1-aixbzbmezu.chromatic.com/?path=/story/all-components-legacycard--with-destructive-action), but I traced the divergence from prod to [this Button PR](https://github.com/Shopify/polaris/pull/10090/files#diff-e545323890f046bc43db9c0fb8cf1cbe8851f075b915fc7295a387e005b92687)). 

I think with the button variant name change `ButtonGroup` doesn't apply the correct item styles anymore 🤔

**Comment by sophschneider** [comment]
I think its [this file](https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/ButtonGroup/components/Item/Item.tsx#L21) that needs to be updated! I'll do it in a separate PR :D

**Pull request merged**